daemon/logger/syslog/syslog.go

```diff
@@ -0,0 +1,61 @@
+package syslog
+
+import (
+	"log/syslog"
+	"os"
+	"path"
+	"sync"
+
+	"github.com/docker/docker/daemon/logger"
+)
+
+type Syslog struct {
+	writer *syslog.Writer
+	tag    string
+	mu     sync.Mutex
+}
+
+func New(tag string) (logger.Logger, error) {
+	return &Syslog{
+		writer: nil,
+		tag:    tag,
+	}, nil
+}
+
+func (s *Syslog) Log(msg *logger.Message) error {
+	s.mu.Lock()
```

Comment:
@LK4D4 looking at the source, we don't need to lock on writes because the syslog code already locks.   But if New fails because Dial returns an error (easiest way to reproduce is to delete /dev/log) then the container will fail to start.  I think we should still lazily initialize the logger while holding a lock.

If we get an error on write, there is no need to close/nilify the writer because syslog code will reconnect later.